src/ray/core_worker/test/core_worker_test.cc

```diff
@@ -0,0 +1,880 @@
+        RAY_CHECK_OK(driver.Objects().Get({object_ids[1]}, -1, &results));
+        // Acotr is created, wait for the gcs notification to propogate to core worker.
+        sleep(2);
+
+        RAY_LOG(INFO) << "actor has been reconstructed";
+      }
+
+      // wait for actor being reconstructed.
+      std::vector<uint8_t> arg1(dis(gen), value_dis(gen));
+      auto buffer1 = std::make_shared<LocalMemoryBuffer>(arg1.data(), arg1.size());
+
+      // Create arguments with PassByRef and PassByValue.
+      std::vector<TaskArg> args;
+      args.emplace_back(TaskArg::PassByValue(buffer1));
+
+      TaskOptions options{1, resources};
+      std::vector<ObjectID> return_ids;
+      RayFunction func{ray::Language::PYTHON, {}};
+
+      auto status =
+          driver.Tasks().SubmitActorTask(*actor_handle, func, args, options, &return_ids);
+      RAY_CHECK_OK(status);
+      RAY_CHECK(return_ids.size() == 1);
+      // Verify if it's expected data.
+      std::vector<std::shared_ptr<RayObject>> results;
+
+      RAY_CHECK_OK(driver.Objects().Get(return_ids, -1, &results));
+      RAY_CHECK(results[0]->GetData()->Size() == buffer1->Size()) << i;
+      RAY_CHECK(memcmp(results[0]->GetData()->Data(), buffer1->Data(), buffer1->Size()) ==
+                0)
+          << i;
+    }
+  }
+}
+
+void CoreWorkerTest::TestActorFailure(
+    const std::unordered_map<std::string, double> &resources, bool is_direct_call) {
+  CoreWorker driver(WorkerType::DRIVER, Language::PYTHON, raylet_store_socket_names_[0],
+                    raylet_socket_names_[0], JobID::FromInt(1), gcs_options_, nullptr);
+
+  std::unique_ptr<ActorHandle> actor_handle;
+
+  // Test creating actor.
+  std::vector<ObjectID> object_ids;
+  std::vector<std::string> object_id_strs;
+  for (int i = 0; i < 2; i++) {
+    object_ids.push_back(ObjectID::FromRandom());
+    object_id_strs.push_back(object_ids.back().Binary());
+  }
+
+  {
+    // Test creating actor.
+    RayFunction func{ray::Language::PYTHON, {c_test_actor_failover_str}};
+
+    std::vector<TaskArg> args;
+    for (const auto &entry : object_id_strs) {
+      auto object_id_buffer = std::make_shared<LocalMemoryBuffer>(
+          reinterpret_cast<uint8_t *>(const_cast<char *>(entry.data())), entry.size());
+      args.emplace_back(TaskArg::PassByValue(object_id_buffer));
+    }
+
+    ActorCreationOptions actor_options{0 /* disable reconstruction */, is_direct_call,
+                                       resources};
+
+    // Create an actor.
+    RAY_CHECK_OK(driver.Tasks().CreateActor(func, args, actor_options, &actor_handle));
+  }
+
+  // Test submitting some tasks with by-value args for that actor.
+  {
+    std::random_device rd;   // Will be used to obtain a seed for the random number engine
+    std::mt19937 gen(rd());  // Standard mersenne_twister_engine seeded with rd()
+    std::uniform_int_distribution<> dis(1, 10);
+    std::uniform_int_distribution<> value_dis(1, 255);
+    const int num_tasks = 3000;
```

Comment:
why do we need to send so many tasks here?

Reply:
basically to make sure that some tasks are submitted after actor is actually killed.